aptos-move/block-executor/src/view.rs

```diff
@@ -1216,12 +1216,25 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                                 key
                             )))
                         },
-                    );
+                    ),
+                    Ok(None) => Some(Err(code_invariant_error(format!(
+                        "Cannot find metadata op for the group read {:?}",
+                        key
+                    )))),
+                    Err(e) => {
+                        if e.major_status() == StatusCode::SPECULATIVE_EXECUTION_ABORT_ERROR {
```

Comment:
See my comment, I think here we can check that captured reads is in speculative failure state instead of checking e which has this error both in incorrect use and speculative failure cases.